src/debug/createdump/crashinfo.cpp

```diff
@@ -439,13 +439,21 @@ CrashInfo::GetDSOInfo()
         if (!ReadMemory(linkMapAddr, &map, sizeof(map))) {
             return false;
         }
-        char moduleName[257] = { 0 };
+        ArrayHolder<char> moduleName = new char[MAX_LONGPATH];
+        moduleName[0] = '\0';
         if (map.l_name != nullptr) {
-            if (!ReadMemory(map.l_name, &moduleName, sizeof(moduleName) - 1)) {
-                return false;
+            for (int i = 0; i < MAX_LONGPATH; i++)
```

Comment:
The MAX_LONGPATH seems still pretty short - it is defined as 1024. PATH_MAX defined in Unix headers is better - it is 4096. 
A nit - you might read the whole block at once, starting at MAX_LONGPATH, if that fails going to MAX_LONGPATH / 2 etc. That would give you a logarithmic number of calls to ReadMemory (w.r.t. to the max buffer size) instead of linear (w.r.t. to the length of the filename) ones. It seems it would be less calls in 99% of cases.